README.md

```diff
@@ -49,7 +49,27 @@ Contains the JSON schemas used to validate the sources in the integration tests.
 
 ## GTFS Realtime Data Structure
 
-This is still under discussion. [Please comment to share your thoughts](https://github.com/MobilityData/mobility-database-catalogs/issues/36). [You can view what is currently drafted in the working document](https://docs.google.com/document/d/1Mlz3AXHItInitsOEAKKi8hZV9d3t3gpFbyIZGm67ESU/edit#heading=h.i8ficnh0oxru).
+|Field Name|Type|Presence|Definition|                                                                     
+|-|-|-|-|
+| mdb_source_id |  Unique ID | System generated | Unique numerical identifier.      |
+| data_type     | Enum| Required| The data format that the source uses: gtfs or gtfs-rt.                                                                                                |
+| provider     | Text | Required                   | A commonly used name for the transit providers included in the source.  
```

Comment:
Should/must this match the provider listed in the listed `mdb_source_id`?

Same for `name` below

Reply:
@barbeau Do you mean the provider listed in the associated `static_reference`? That could be enforced. Name would be unique to the actual realtime source and its ID, e.g "consolidated realtime for the Bay Area"

Reply:
You're correct about how the "source" record would work based on the current structure. Thanks for bringing in a concrete example and making it clear we need to define this better. TransitFeeds groups GTFS schedule sources similarly to what we're doing right now, in that 1 feed = 1 download URL. The difference is that the user interface on TransitFeeds makes it possible to search by provider, so you can see a listing of all feeds under 1 provider before clicking into each feed. 

@maximearmstrong and I were discussing modifying this so that there's 1 download URL per source for GTFS Realtime as well. This would make the definition of a source more consistent across data types - 1 downloadable stream of data - and ensure we're creating a data model that's more scalable over time as we build out a SQL database. It's clear that there's a need for [the providers catalog](https://portal.productboard.com/8wemmnsxdznjivwrft7wh1wp/c/231-organizations-and-services-catalogs) and an easy way to query and search for all sources by provider, but that's out of scope for the particular focus of the GTFS realtime source schema.

You and @e-lo's thoughts on this would be extremely helpful here. I think @e-lo proposed something similar to this at the beginning of the schema discussion, with us modifying the `data-type` to specify which specific realtime type is included in a source.

Reply:
>@maximearmstrong and I were discussing modifying this so that there's 1 download URL per source for GTFS Realtime as well. This would make the definition of a source more consistent across data types - 1 downloadable stream of data - and ensure we're creating a data model that's more scalable over time as we build out a SQL database.

I think the difference between JSON and SQL is part of the issue here. JSON is de-normalized (typically) by design, and SQL is normalized. So there is tension between the data model (which is independent of implementation) and it's expression (JSON or SQL).

If you're representing the data in JSON now, I'm strongly in favor of keeping all the GTFS RT URLs in the same file under the same source ID. If you want to give each RT URL endpoint it's own ID to use as a primary key in a SQL database later, then add a new ID field to that object in the schema (you could actually do the same for static feed endpoints, so they have IDs separate from the source IDs).

If you de-normalized the RT feed endpoints across multiple JSON files IMHO that's making life a lot more complicated for consumers, who would then have to reconstruct this list themselves into a new data model by looping through the datasets multiple times.

Reply:
@barbeau A part of what's making us reconsider the model is that all consumers we've talked to who are using the catalogs right now are relying on [the CSV](https://storage.googleapis.com/storage/v1/b/mdb-csv/o/sources.csv?alt=media) and operations to pull the URLs they need rather than the JSON files. If we were to represent the schema with multiple download URLs in the CSV, [it would look like this using your original Hillsborough example](https://docs.google.com/spreadsheets/d/1H1Ccn1vPK6DWt6v6TUA2ax9IoZA6QPGmB9Ex_xY307U/edit?usp=sharing). 

This seems more difficult to search (since data-type is `gtfs-rt` for all of them) and means that 1 row no longer = 1 source in the sheet. It'll also make it harder to add and update sources using the operations given the length of the schema. 

Based on these prospective pain points for current users, we're leaning towards 1 download URL per source. We're aiming to finalize this schema before the end of the week, so we're looking for feedback over the next two days to help inform the final decision before I apply the other minor changes requested here.

Reply:
>A part of what's making us reconsider the model is that all consumers we've talked to who are using the catalogs right now are relying on [the CSV](https://storage.googleapis.com/storage/v1/b/mdb-csv/o/sources.csv?alt=media) and operations to pull the URLs they need rather than the JSON files.

Just curious - are these consumers that are using the CSV files humans that are manipulating data in spreadsheets, or software that's parsing the data and doing operations on that data?

If humans, then I think we're talking about very different use cases/audiences. I'm looking at this primarily from the perspective of wanting to write software to parse the data like we did with TransitFeeds.com in https://github.com/CUTR-at-USF/transit-feed-quality-calculator to produce global GTFS RT feed quality metrics.

>This seems more difficult to search (since data-type is gtfs-rt for all of them)

Not sure I understand this - would the data-type change to something like `trip-updates` in the model you're proposing? In other words, why wouldn't it be `gtfs-rt`?

>It'll also make it harder to add and update sources using the operations given the length of the schema.

Not sure I understand this either - could you elaborate?

>Based on these prospective pain points for current users, we're leaning towards 1 download URL per source.

I understand why you would make this choice, especially if it's what the current users want, and I agree that it does match the static schema more directly. However, just my two cents - I think this also kicks the can down the road to tackle the questions like what a "source" really it, and how to group multiple feeds under the same source or provider. In other words, the limitations of the static schema are also being adopted in the RT schema. In the meantime, this responsibility is pushed back on consumers to make assumptions and stitch the records together themselves, which I think will hinder adoption for those use cases. Maybe these aren't seen as limitations by the current user base, but from my perspective, it seems like a step backwards from what the TransitFeeds.com API offered, from what I recall. 

All that being said, I suppose this could all be a v1 of the schema. In the future you could also introduce new ID fields down the road to group records by provider/source, and update the records at that point with a new JSON format for a schema v2 to make consuming these records easier. In the meantime, I think any clearly written narrative guidance and examples that you could provide for how real-world use cases are treated (e.g., HART, MTA NYC examples) would reduce the burden on existing consumers to understand how these cases fit into this version of the schema.

Reply:
> > This seems more difficult to search (since data-type is gtfs-rt for all of them)
> 
> Not sure I understand this - would the data-type change to something like `trip-updates` in the model you're proposing? In other words, why wouldn't it be `gtfs-rt`?

If you look at the example [here] (https://docs.google.com/spreadsheets/d/1H1Ccn1vPK6DWt6v6TUA2ax9IoZA6QPGmB9Ex_xY307U/edit?usp=sharing), the three rows share the same `mdb_source_id` and the same `data_type = gtfs-rt`, but the URLs are not the same in the three rows. At first, the rows appear as duplicates, but they are not. We thought of solving this problem by using arrays of objects for the Realtime URLs, but that means we would have to store JSON objects in a CSV, which doesn't seem ideal for the end user.

If we do 1 file 1 URL, then we would have a different `mdb_source_if` for each row.

> > It'll also make it harder to add and update sources using the operations given the length of the schema.
> 
> Not sure I understand this either - could you elaborate?

We provide [Python functions](https://github.com/MobilityData/mobility-database-catalogs#using-the-mobility-database-catalogs) to help users add or update sources in the catalogs. Currently, adding a GTFS Schedule source is fairly easy, as there is only one `direct_download_url` per source. If we update the GTFS Realtime JSON schema to allow multiple URLs per source, adding a GTFS Realtime source using a function will become more complicated for the user (creating a list of objects to pass to the function, etc.). We could provide more functions to help them, but it probably wouldn't be as easy.

> > A part of what's making us reconsider the model is that all consumers we've talked to who are using the catalogs right now are relying on [the CSV](https://storage.googleapis.com/storage/v1/b/mdb-csv/o/sources.csv?alt=media) and operations to pull the URLs they need rather than the JSON files.
> 
> Just curious - are these consumers that are using the CSV files humans that are manipulating data in spreadsheets, or software that's parsing the data and doing operations on that data?

I couldn't say for other users, but most of the time I use CSV in scripts that are outside the repository, mainly because we don't have an API at the moment. 

Initially, our idea for the catalogs was to store all sources in a CSV, similar to `systems.csv` for GBFS. We then realized that a CSV would be difficult to extend with additional fields, but at the same time it is more human readable.
This led us to choose JSON files to store the sources, so that it is easier to manipulate the data in the repository and validate it against a schema, knowing that we would still export as a CSV so that people could easily discover the sources.

Using the CSV outside of the repository is generally easier right now, because it contains all the sources and we export it to a GCP bucket with a stable URL. Ideally, we would develop an API as a long-term alternative.
Does this make sense?

Reply:
>Does this make sense?

Yes, I understand the above. It sounds like your current use cases are different than the one I'm most familiar with (JSON as a API, like the TransitFeeds.com API), so I'll defer to your judgement for what you think meets your users needs the best.

Reply:
> Yes, I understand the above. It sounds like your current use cases are different than the one I'm most familiar with (JSON as a API, like the TransitFeeds.com API), so I'll defer to your judgement for what you think meets your users needs the best.

Oh, I totally agree that in terms of API responses, your approach is much better. I think our ultimate goal would be to provide an additional API layer, running on a server, that would use the catalogs as a source of truth and return JSON responses formed similarly to your schema or the TransfitFeeds one. Thank you for all the feedback you have provided, I think it is very valuable for the API planning.